Code/Tools/ProjectManager/Source/CreateAGemScreen.cpp

```diff
@@ -0,0 +1,446 @@
+        repositoryURLErrorLabel->setObjectName("createAGemFormErrorLabel");
+        QLineEdit* repositoryURLLineEdit = m_repositoryURL->lineEdit();
+        repositoryURLLineEdit->setPlaceholderText(tr("https://github.com/Jane"));
+        thirdScreen->addWidget(m_repositoryURL);
+
+        return thirdScreenScrollArea;
+    }
+
+    bool CreateAGemScreen::ValidateGemDisplayName()
+    {
+        bool gemScreenNameIsValid = true;
+        if (m_tabWidget->currentIndex() == 2)
+        {
+            if (m_gemDisplayName->lineEdit()->text().isEmpty())
+            {
+                gemScreenNameIsValid = false;
+                m_gemDisplayName->setErrorLabelText(tr("A gem display name is required."));
+            }
+
+            m_gemDisplayName->setErrorLabelVisible(!gemScreenNameIsValid);
+        }
+        return gemScreenNameIsValid;
+    }
+
+    bool CreateAGemScreen::ValidateGemSystemName()
+    {
+        bool gemSystemNameIsValid = true;
+        if (m_tabWidget->currentIndex() == 2)
+        {
+            if (m_gemSystemName->lineEdit()->text().isEmpty())
+            {
+                gemSystemNameIsValid = false;
+                m_gemSystemName->setErrorLabelText(tr("A gem system name is required."));
+            }
+
+            m_gemSystemName->setErrorLabelVisible(!gemSystemNameIsValid);
+        }
+        return gemSystemNameIsValid;
+
+    }
+
+    bool CreateAGemScreen::ValidateLicenseName()
+    {
+        bool licenseNameIsValid = true;
+        if (m_tabWidget->currentIndex() == 2)
+        {
+            if (m_license->lineEdit()->text().isEmpty())
+            {
+                licenseNameIsValid = false;
+                m_license->setErrorLabelText(tr("License details are required."));
+            }
+
+            m_license->setErrorLabelVisible(!licenseNameIsValid);
+        }
+        return licenseNameIsValid;
+    }
+
+    bool CreateAGemScreen::ValidateCreatorName()
+    {
+        bool creatorNameIsValid = true;
+        if (m_tabWidget->currentIndex() == 2)
+        {
+            m_numNextClicks += 1;
+            if (m_numNextClicks > 1)
+            {
+                if (m_creatorName->lineEdit()->text().isEmpty())
+                {
+                    creatorNameIsValid = false;
+                    m_creatorName->setErrorLabelText(tr("A creator's name is required."));
+                }
+
+                m_creatorName->setErrorLabelVisible(!creatorNameIsValid);
+            }
+        }
+        return creatorNameIsValid;
+    }
+
+    bool CreateAGemScreen::ValidateRepositoryURL()
+    {
+        bool repositoryURLIsValid = true;
+        if (m_tabWidget->currentIndex() == 2)
+        {
+            if (m_numNextClicks > 1)
+            {
+                if (m_repositoryURL->lineEdit()->text().isEmpty())
+                {
+                    repositoryURLIsValid = false;
+                    m_repositoryURL->setErrorLabelText(tr("A repository URL is required."));
+                }
```

Comment:
this should only validate the URL if it is not empty and right now this only checks if one exists.